### PR TITLE
Upgrade mingw-w64 on CI

### PR DESCRIPTION
**Pull request by mati865**
Continuation of https://github.com/rust-lang/rust/pull/99162

This will require copying related archives mentioned in https://github.com/rust-lang/rust/pull/100178#issuecomment-1352060140 to Rust's mirror.

This should fix multiple issues with currently used ancient releases but I'll ask folks to retest them using nightly once it goes live.

### CODE REVIEW

File: src/ci/scripts/install-mingw.sh
Comment (rylev) [inline src/ci/scripts/install-mingw.sh:30]: I'll leave a note here that the comment at the top of this file will need to be updated should this land.

File: src/ci/scripts/install-mingw.sh
Comment (klensy) [inline src/ci/scripts/install-mingw.sh:12]:
```suggestion
MINGW_ARCHIVE_32="i686-12.1.0-release-posix-dwarf-rt_v10-rev3.7z"
```

**Comment by petrochenkov** [comment]
So I wonder maybe we should use a slightly older version of mingw on CI, if possible (as long as it supports removing `llvm.allow-old-toolchain`).
The newest version is already tested by anyone using MSYS2 and updating packages regularly (e.g. me), but some older version support may accidentally break without a CI check.

**Comment by mati865** [comment]
It's not easy to obtain them: we would have to build them ourselves.
Also there are issues like https://github.com/rust-lang/rust/issues/88704 that have been fixed in recent GCC versions (it shouldn't appear with GCC 11/12) but I have no clue how old compiler we could built and still have that fix.

**Comment by petrochenkov** [comment]
niXman's github has this oldest release - https://github.com/niXman/mingw-builds-binaries/releases/tag/11.2.0-rt_v9-rev1, that may be good enough.

**Comment by mati865** [comment]
That didn't work out at all:
```
  = note: D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/lib/../lib/crt2.o:crtexe.c:(.rdata$.refptr.mingw_app_type[.refptr.mingw_app_type]+0x0): undefined reference to `mingw_app_type'
          D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/lib/../lib/crt2.o:crtexe.c:(.rdata$.refptr.mingw_initcharmax[.refptr.mingw_initcharmax]+0x0): undefined reference to `mingw_initcharmax'
          D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/lib/../lib/crt2.o:crtexe.c:(.rdata$.refptr.mingw_initltssuo_force[.refptr.mingw_initltssuo_force]+0x0): undefined reference to `mingw_initltssuo_force'
          D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/lib/../lib/crt2.o:crtexe.c:(.rdata$.refptr.mingw_initltsdyn_force[.refptr.mingw_initltsdyn_force]+0x0): undefined reference to `mingw_initltsdyn_force'
          D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/lib/../lib/crt2.o:crtexe.c:(.rdata$.refptr.mingw_initltsdrot_force[.refptr.mingw_initltsdrot_force]+0x0): undefined reference to `mingw_initltsdrot_force'
          collect2.exe: error: ld returned 1 exit status
```
So upgraded to newer GCC 12.2 release instead and tested again both 32-bit and 64-bit builds locally.

**Comment by pietroalbini** [comment]
Likely won't have time in the near future to look into this. r? @Mark-Simulacrum

**Comment by Mark-Simulacrum** [comment]
I don't have any real opinion on whether to go for older or more recent release binaries, but am happy to mirror binaries as needed when we're ready to do so. Will leave this as waiting on author for now though to figure out a path toward a clear answer for which binaries to copy up.

**Comment by mati865** [comment]
Binaries with GCC 11 and older mingw-w64 didn't work and upon inspection tarballs called GCC 12.1 and GCC 12.2 differ only in GCC version (mingw-w64 and Binutils are the same) so I'd go with GCC 12,2 since it carries mostly bugfixes over GCC 12.1 (so there are hopefully less chances for issues with things like building LLVM).

@rustbot ready

**Comment by Mark-Simulacrum** [comment]
Uploaded the binaries in question to the mirror -- so I think from an infra perspective this should be good to go.

I'm marking this as relnotes since it sounds (https://github.com/rust-lang/rust/pull/100178#issuecomment-1229021255) like this could break some workflows; along those lines I wonder if it makes sense to have this version be listed as part of the platform support page? Currently that just says MinGW -- should it have a version?

r? @wesleywiser perhaps to confirm the policy angle here vis-a-vis platform support (mostly since you might have more Windows context than me).

**Comment by mati865** [comment]
I worry there is only one way to find out if this breaks some workflows.

> I'm marking this as relnotes since it sounds (https://github.com/rust-lang/rust/pull/100178#issuecomment-1229021255) like this could break some workflows;

Maybe it would be worth to call for testing or something like that?

> along those lines I wonder if it makes sense to have this version be listed as part of the platform support page? Currently that just says MinGW -- should it have a version?

It should at very least say to use mingw-w64 and Dwarf-2/SEH exception models for i686/x86_64 respectively.

Regarding the version, declaring GCC 12 and mingw-w64 10 as lowest supported versions would be very restrictive but those are the versions that would be tested on CI.
Staying on GCC 6 is not an option because LLVM 16 will remove option that allows it to be built with GCC older than 7.1.

**Comment by wesleywiser** [comment]
I have to admit, I'm not super familiar with the details of the `windows-gnu` targets. 

What sort of workflows do we think could theoretically break? 

Does this have any effect on the minimum supported version of Windows for these targets?

File: src/test/debuginfo/embedded-visualizer.rs
Comment (mati865) [reply src/test/debuginfo/embedded-visualizer.rs:4]: It's built with Python support but I have no clue about using it, all I have is this output confirming it didn't work for some reason:
```
Thread 1 hit Breakpoint 1, embedded_visualizer::main () at src/test\debuginfo/embedded-visualizer.rs:107
107         zzz(); // #break
No auto-load scripts.
$1 = embedded_visualizer::point::Point {x: 0, y: 0}
$2 = embedded_visualizer::point::Point {x: 5, y: 8}
$3 = embedded_visualizer::Line {a: embedded_visualizer::point::Point {x: 0, y: 0}, b: embedded_visualizer::point::Point {x: 5, y: 8}}
$4 = dependency_with_embedded_visualizers::Person {name: alloc::string::String {vec: alloc::vec::Vec<u8, alloc::alloc::Global> {buf: alloc::raw_vec::RawVec<u8, alloc::alloc::Global> {ptr: core::ptr::unique::Unique<u8> {pointer: core::ptr::non_null::NonNull<u8> {pointer: 0x2af52f02ca0}, _marker: core::marker::PhantomData<u8>}, cap: 8, alloc: alloc::alloc::Global}, len: 8}}, age: 10}
```

**Comment by mati865** [comment]
> What sort of workflows do we think could theoretically break?

Hopefully the only visible change (apart from numerous fixes) will be for people who rely on `rust-mingw` component instead of using external toolchain: I imagine static libs built with that setup would no longer work with old toolchains but in turn they would start working with new toolchains.
I expect this situation to be very rare as most of the users have external mingw-w64 toolchain installed in their system.

> Does this have any effect on the minimum supported version of Windows for these targets?

Apparently mingw-w64 has increased default Windows version to Windows 10 [last year](https://gitlab.com/mati865/mingw-w64-mirror/-/commit/f3c53a51df5c08f181e13a39b1cd6fd1d41edb96). So it might not work with Windows 7 which I cannot easily test.

File: src/test/debuginfo/embedded-visualizer.rs
Comment (wesleywiser) [reply src/test/debuginfo/embedded-visualizer.rs:4]: Weird... I think it's fine to just leave a comment that embedded Python visualizations don't work on `windows-gnu` without investigating further.

Comment (petrochenkov) [reply src/test/debuginfo/embedded-visualizer.rs:4]: IIRC python visualizers are just not embedded into binaries on windows-gnu (`emit_debug_gdb_scripts: false`).
And the test didn't previously run on windows-gnu CI due to the `min-gdb-version: 8.1` requirement.

**Comment by wesleywiser** [comment]
> people who rely on rust-mingw component instead of using external toolchain

Ah so people that use mingw components we distribute with the toolchain instead of using the same components from mingw itself? That feels analogous to the LLVM tools we ship on various platforms. Users could rely on those and we might break their workflow during an LLVM upgrade but that isn't really a supported scenario. I think from that perspective, my concern is resolved. 

> Apparently mingw-w64 has increased default Windows version to Windows 10 [last year](https://gitlab.com/mati865/mingw-w64-mirror/-/commit/f3c53a51df5c08f181e13a39b1cd6fd1d41edb96). So it might not work with Windows 7 which I cannot easily test.

This is more concerning to me since `{i686,x86_64}-pc-windows-gnu` are Tier 1 platforms which claim to support Windows 7+. I think we will probably need to go through a similar process as we did in #95026 which raised the platform requirements for the Linux targets.

**Comment by cuviper** [comment]
This is going to be more urgent when we start the LLVM 16 upgrade, which will _require_ at least GCC 7.1 (i.e. no more option to allow-old-toolchain here). If we can avoid the Windows 7 question for now by choosing an older build, or building our own, I think that should be strongly considered.

**Comment by mati865** [comment]
> This is going to be more urgent when we start the LLVM 16 upgrade, which will require at least GCC 7.1 (i.e. no more option to allow-old-toolchain here).

Yep, that was the main motivation for me to look at it again.

> If we can avoid the Windows 7 question for now by choosing an older build

For some various reasons older builds do not work for Rust.

> building our own

Scripts used to build toolchain used in this PR are available at https://github.com/niXman/mingw-builds, building from scratch is also possible albeit annoying with some outdated guides like https://sourceforge.net/p/mingw-w64/wiki2/Cross%20Win32%20and%20Win64%20compiler/.

I think there is no defined process on how to handle this though.

**Comment by cuviper** [comment]
Could we use the toolchain from MSYS2? They're in the process of phasing out Windows 7 too, but maybe we can take builds from before they started those changes?

**Comment by cuviper** [comment]
> I guess one could install minimal set of toolchain packages (Binutils + GCC + GDB + mingw-w64) and simply tar that directory. Then use it the same way as we install other toolchains.

Yeah, I was thinking it would be something like that.

> There are some downsides to that approach like mingw-w64 version. MSYS2 uses its latest git revision (for various good reasons that do not matter for Rust) which would mean CI would test compatibility only with very very CRT version that should be released around spring next year.

We do recommend MSYS2 in the README though, so it seems nicer to build and test with that rather than a 3rd-party build. Are there specific version-compatibility changes you're concerned about?

**Comment by ChrisDenton** [comment]
Hm, to be clear, we don't claim to support compiling Rust itself on Windows 7? Like it may or may not work but we don't make any promises.

**Comment by mati865** [comment]
> We do recommend MSYS2 in the README though, so it seems nicer to build and test with that rather than a 3rd-party build. Are there specific version-compatibility changes you're concerned about?

Rust provides `rust-mingw` component which bundles mingw-w64 static and import libraries and binaries like GCC and ld.bfd from build machine. I don't know if it's case that Rust wants to support (I really hope it's not the case) but if user took Rust 1.64 and used bundled `rust-mingw` component to create static libraries (assuming it's possible?) and then used that static library while linking some binary they could notice it fails with nightly after this merge. Since MSYS2 uses unreleased mingw-w64 version there might be impossible for them to use matching toolchain but again this is already super fragile to rely on.

Another thing that came to my mind is debuginfo, old toolchains like one shipped right now default to dwarf-2 but recent versions use Dwarf 4 (IIRC it was bumped to version 5 and then downgraded to version 4 but I could be wrong). It's unclear to me how it affects compatibility with old toolchains.

> Hm, to be clear, we don't claim to support compiling Rust itself on Windows 7? Like it may or may not work but we don't make any promises.

Dunno but this change may break running Rust on Windows 7 if LLVM or compatibility layers like winpthreads start using some new API because `_WIN32_WINNT` tells them it's fine.

**Comment by mati865** [comment]
How should we move forward?
The longer it takes the less time is left to iron out eventual issues before LLVM 16 testing starts.

**Comment by cuviper** [comment]
I'm not a stakeholder for this target, but it seems to me the options are:

1. Use the niXman build and drop support for Windows 7. That's a Tier-1 support change, which we've been leaning toward already, but could take a while for approval.
2. Use a snapshot of the MSYS2 toolchain. We might have to grab a build before [this hardware change](https://www.msys2.org/news/#2022-10-18-new-minimum-hardware-requirements-cpus-from-20067), if that's even available, or else risk that our builds become slightly less compatible.
3. Build our own toolchain with whatever compat settings we want. This may be a lot of effort, but at least it's infrequent...

**Comment by mati865** [comment]
I don't feel like the person to take decision here either and I have no idea on how the decision should be taken.

**Comment by cuviper** [comment]
Let's ask the compiler team to consider the compatibility aspect, at least.

@rustbot label +I-compiler-nominated

**Comment by apiraino** [comment]
Visited during the weekly T-compiler meeting, follow-up discussion will happen on this [Zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Upgrading.20MinGW.20on.20the.20CI)

@rustbot label -I-compiler-nominated

**Comment by nikic** [comment]
> Visited during the weekly T-compiler meeting, follow-up discussion will happen on this [Zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Upgrading.20MinGW.20on.20the.20CI)

Is this the right link? The last activity there was in 2020.

**Comment by apiraino** [comment]
> Is this the right link? The last activity there was in 2020.

Yes. T-compiler visiting this PR happened here [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202022-11-10/near/309020772) and was suggested to keep the conversation on that older [Zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Upgrading.20MinGW.20on.20the.20CI). I may set another reminder for the next meeting :)

**Comment by nikic** [comment]
> Build our own toolchain with whatever compat settings we want. This may be a lot of effort, but at least it's infrequent...

So, I looked into how hard it would be to redo the [niXman/mingw-builds](https://github.com/niXman/mingw-builds) builds with a lower WINNT requirement, and it turns out that this is not hard at all. The build script supports a `--with-default-win32-winnt` option, so adjusting the GA workflow to pass `--with-default-win32-winnt=0x0601` would produce a Windows 7 compatible toolchain.

And while looking into this, I found that we're not the first people who want this... https://github.com/starg2/mingw-builds/commit/316ee4370be86d9936dac95416d247e608616c22 is the necessary change to the GA workflow, and https://github.com/starg2/mingw-builds/actions/runs/3350177834 are the build artifacts for that commit.

I think we can just go ahead and grab those builds?

**Comment by mati865** [comment]
> I think we can just go ahead and grab those builds?

I guess so.
While I don't expect any surprises here I can run tests locally if the reviewer wants.

**Comment by nikic** [comment]
> > I think we can just go ahead and grab those builds?
> 
> I guess so. While I don't expect any surprises here I can run tests locally if the reviewer wants.

Probably wouldn't hurt?

**Comment by nikic** [comment]
Actually, I think we should just go ahead and merge this in the current form. LLVM does define `_WIN32_WINNT=0x0601` (e.g. in https://github.com/llvm/llvm-project/blob/21d3871b7c90f85b3ae417724d7864369173bde0/llvm/include/llvm/Support/Windows/WindowsSupport.h#L29 for the core part, but also in other places), so in theory it should be fine.

As the platform support pages says:

> Only Windows 10 currently undergoes automated testing. Earlier versions of Windows rely on testing and support from the community.

It doesn't look like anyone is able or willing to test whether this works on Windows 7 to say for sure, so we should rely on that community testing. If it turns out that it doesn't work we already have potential avenues to address that, and will also have someone who can actually confirm whether or not new builds resolve any encountered issue.

**Comment by cuviper** [comment]
> LLVM does define `_WIN32_WINNT=0x0601` (...), so in theory it should be fine.

Don't we also need to worry about pre-compiled CRT bits? I would expect that to be highly affected by the OS level.

**Comment by nikic** [comment]
@cuviper I've been told that the mingw-w64 crt should not be affected. `--with-default-win32-winnt` is a configuration option for `mingw-w64-headers` only and it should be possible to target older versions as long as `_WIN32_WINNT` is explicitly set.

**Comment by cuviper** [comment]
@wesleywiser you're the assigned reviewer -- how do you feel about that?

File: src/ci/scripts/install-mingw.sh
Comment (nikic) [inline src/ci/scripts/install-mingw.sh:68]: Don't we need to keep this code, apart from the gdb replacement? Otherwise we're not actually performing the download anywhere?

**Comment by mati865** [comment]
Resolved the conflicts and tested manually with https://github.com/rust-lang/rust/pull/100178#issuecomment-1352060140 (sorry it took so long but setting this up is easier said than done).

There is one new failing UI test that I added to the ignored:
`error: line not found in debugger output: [...]$1 = 11` it's because on the masterGDB is older than `min-gdb-version: 8.1` and we do not have pretty printers: `$1 = core::num::nonzero::NonZeroI8 (11)`.

I have some unexpected failures locally on both MSYS2 toolchain and the mentioned toolchain but I don't think they will appear on the CI:
```
failures:
    fs::tests::create_dir_all_with_junctions
    fs::tests::recursive_rmdir
    fs::tests::recursive_rmdir_of_symlink
```

File: src/ci/scripts/install-mingw.sh
Comment (mati865) [reply src/ci/scripts/install-mingw.sh:68]: Honestly I have no clue what happened there but you are right.

**Comment by nikic** [comment]
Thanks! @Mark-Simulacrum Could you please re-upload the `i686-12.2.0-release-posix-dwarf-rt_v10-rev0.7z` and `x86_64-12.2.0-release-posix-seh-rt_v10-rev0.7z` builds from https://github.com/starg2/mingw-builds/actions/runs/3350177834?

**Comment by wesleywiser** [comment]
> @wesleywiser you're the assigned reviewer -- how do you feel about that?

Just to make sure I understand the current state, it looks like we've figured out how to build mingw with a recent gcc (12) and still retain Windows 7 compatibility? As such we don't need to change the supported version of the `*-windows-gnu` targets and we can unblock the LLVM upgrade. 

Assuming that's correct, I'm happy to r+! I don't have a ton of experience with `*-windows-gnu` but I trust @mati865's testing and expertise with the target 🙂

**Comment by mati865** [comment]
> Just to make sure I understand the current state, it looks like we've figured out how to build mingw with a recent gcc (12) and still retain Windows 7 compatibility? As such we don't need to change the supported version of the `*-windows-gnu` targets and we can unblock the LLVM upgrade.

Yeah, nikic has found it.

> Assuming that's correct, I'm happy to r+! I don't have a ton of experience with *-windows-gnu but I trust @mati865's testing and expertise with the target 🙂

While I think (and hope!) there will be no breakage I wouldn't rollup it personally.

**Review by wesleywiser — APPROVED** [review]
r=me when the infra side is ready

**Comment by Mark-Simulacrum** [comment]
Both files should be uploaded, e.g., at https://ci-mirrors.rust-lang.org/rustc/x86_64-12.2.0-release-posix-seh-rt_v10-rev0.7z

**Comment by nikic** [comment]
Thanks! Let's wait for https://github.com/rust-lang/rust/pull/107137 to land before merging this, so it doesn't go in at the very end of the cycle.

**Comment by mati865** [comment]
I'm having hard time reproducing the failure from CI because after I rebased this branch on the master and applied workaround for https://github.com/rust-lang/rust/issues/105696 (disabled `git` detection) I'm unable to run tests for any target:
```
$ ./x.py t --build i686-pc-windows-gnu
Building rustbuild
   Compiling bootstrap v0.0.0 (D:\Projekty\rust\src\bootstrap)
    Finished dev [unoptimized] target(s) in 5.29s
Build completed successfully in 0:00:09
```

**Comment by cuviper** [comment]
I was able to reproduce the three `run-make` failures:

```
    [run-make] tests\run-make\raw-dylib-inline-cross-dylib
    [run-make] tests\run-make\rlib-format-packed-bundled-libs
    [run-make] tests\run-make\rlib-format-packed-bundled-libs-2
```

These are all trying to grep output from `llvm-objdump` and `llvm-nm`. When I change them to use mingw's binutils instead, like the one you already changed due to "wonky" piping, then they pass.

**Comment by nikic** [comment]
Do we know what is actually wrong with the piping in the LLVM tools? If this is something we can fix on the LLVM side, would be nice to do that for the LLVM 16 release.

**Comment by mati865** [comment]
@cuviper oh, nice. I've been trying to find out why bootstrap stopped working for me (on Win 11 machine) multiple times but mode no further progress so far.

@nikic IIRC this has been an issue with LLVM since it's been packaged so I assume it has never worked.
I don't know if it's an issue with Cygwin or LLVM but with GNU tools there was never such an issue.
I haven't investigated it myself and doubt I can allocate some time for it.

**Comment by ChrisDenton** [comment]
I'm not sure if it's the cause of this issue but cygwin does a number of nonstandard things to stdio handles which has a tendency to break ordinary windows programs in "fun" ways. Especially with large inputs/outputs.

**Comment by mati865** [comment]
Updated 3 tests that failed on CI to use GNU `nm`.
FWIW I still cannot make successful build with toolchain mentioned in this PR and using MSYS2 i686 toolchain I get many failures that I don't have time to check:
```
failures:
    [run-make] tests\run-make\issue-36710
    [run-make] tests\run-make\native-link-modifier-bundle
    [run-make] tests\run-make\native-link-modifier-whole-archive
    [run-make] tests\run-make\raw-dylib-alt-calling-convention
    [run-make] tests\run-make\raw-dylib-c
    [run-make] tests\run-make\raw-dylib-import-name-type
    [run-make] tests\run-make\raw-dylib-inline-cross-dylib
    [run-make] tests\run-make\raw-dylib-link-ordinal
    [run-make] tests\run-make\raw-dylib-stdcall-ordinal
    [run-make] tests\run-make\rlib-format-packed-bundled-libs
```
So basically just hoping it will pass on CI...

**Comment by nikic** [comment]
Let's try it...

@bors r+ rollup=never

**Comment by cuviper** [comment]
> I'm not sure if it's the cause of this issue but cygwin does a number of nonstandard things to stdio handles which has a tendency to break ordinary windows programs in "fun" ways. Especially with large inputs/outputs.

I wonder if we can interpose something to alleviate that? I'm thinking along the lines of `src/etc/cat-and-grep.sh` that's used in a few tests. e.g. if large outputs are a problem, maybe we can apply different buffering settings.

**Comment by ChrisDenton** [comment]
For sure, it's likely not an insurmountable problem (if that is the issue). But I'm not personally very familiar with cygwin. I just remember this was an issue we worked around in the std so that it wouldn't be UB.

File: tests/run-make/raw-dylib-inline-cross-dylib/Makefile
Comment (cuviper) [inline tests/run-make/raw-dylib-inline-cross-dylib/Makefile:24]: Why did you switch to `nm` instead of `objdump`?

Comment (mati865) [reply tests/run-make/raw-dylib-inline-cross-dylib/Makefile]: I was in hurry and copied the solution form the other test. Should be fine now.

File: tests/run-make/raw-dylib-inline-cross-dylib/Makefile
Comment (nikic) [inline tests/run-make/raw-dylib-inline-cross-dylib/Makefile:13]: Should be `objdump` here?

Though this is an `only-windows` test, so probably having an IS_WINDOWS condition doesn't make a lot of sense.

Comment (mati865) [reply tests/run-make/raw-dylib-inline-cross-dylib/Makefile]: > Should be objdump here?

Windows is case insensitive so either way works but yeah, let's keep it in line with other tests.

> Though this is an only-windows test, so probably having an IS_WINDOWS condition doesn't make a lot of sense.

Good catch.

**Comment by nikic** [comment]
I opened https://github.com/rust-lang/rust/issues/107453 to keep track of the llvm-nm issue.

**Comment by Ravenant1234** [comment]
I am a Rust student, this affects Rust directly for daily use of rust programs or only Rust compilatión of itself?

**Comment by Ravenant1234** [comment]
> I am a Rust student, this affects Rust directly for daily use of rust programs or only Rust compilatión of itself?

Bc i am using Rust in Windows 7 with Msys2 mingw, this update allow to continue to develop with Windows 7?

**Comment by mati865** [comment]
This version still supports Windows 7 so there shouldn't change for you.
However MSYS2 does not support Windows7: https://www.msys2.org/docs/windows_support/

**Pull request merged**
